pkg/controller/bastion/resourcesdefine.go

```diff
@@ -113,14 +113,14 @@ func computeInstanceDefine(opt *Options, bastion *extensionsv1alpha1.Bastion, pu
 	}
 }
 
-func nsgIngressAllowSSH(opt *Options) network.SecurityRule {
+func nsgIngressAllowSSH(ruleName string, destinationAddress string, sourceAddresses []string) network.SecurityRule {
 	return network.SecurityRule{
-		Name: to.StringPtr(NSGIngressAllowSSHResourceName(opt.BastionInstanceName)),
+		Name: to.StringPtr(ruleName),
 		SecurityRulePropertiesFormat: &network.SecurityRulePropertiesFormat{
 			Protocol:                 network.SecurityRuleProtocolTCP,
-			SourceAddressPrefix:      to.StringPtr(strings.Join(opt.CIDRs, ",")),
+			SourceAddressPrefix:      to.StringPtr(strings.Join(sourceAddresses, ",")),
```

Comment:
```go
SourceAddressPrefix: to.StringPtr(strings.Join(sourceAddresses, ",")),
````

this does not seem to work with multiple addresses: `Security rule /subscriptions/00d2caa5-cd29-46f7-845a-2f8ee0360ef5/resourceGroups/shoot--core--ps-bastion-test/providers/Microsoft.Network/networkSecurityGroups/shoot--core--ps-bastion-test-workers/securityRules/shoot--core--ps-bastion-test-cli--bastion-d379b-allow-ssh IPv4 has invalid Address prefix. Value provided: 1.2.3.4/32,3.4.5.6/32\"`

```yaml
apiVersion: extensions.gardener.cloud/v1alpha1
kind: Bastion
metadata:
  finalizers:
  - extensions.gardener.cloud/bastion
  name: cli-az2
  namespace: shoot--core--ps-bastion-test
spec:
  ingress:
  - ipBlock:
      cidr: 1.2.3.4/32
  - ipBlock:
      cidr: 3.4.5.6/32
  type: azure
  userData: IyEvY...
status:
  ingress:
    ip: 7.8.9.10
  lastError:
    description: 'Error reconciling bastion: can''t update Network Security Group
      shoot--core--ps-bastion-test-workers: network.SecurityGroupsClient#CreateOrUpdate:
      Failure sending request: StatusCode=400 -- Original Error: Code="SecurityRuleInvalidAddressPrefix"
      Message="Security rule /subscriptions/00d2caa5-cd29-46f7-845a-2f8ee0360ef5/resourceGroups/shoot--core--ps-bastion-test/providers/Microsoft.Network/networkSecurityGroups/shoot--core--ps-bastion-test-workers/securityRules/shoot--core--ps-bastion-test-cli--bastion-d379b-allow-ssh
      IPv4 has invalid Address prefix. Value provided: 1.2.3.4/32,3.4.5.6/32"
      Details=[]'
    lastUpdateTime: "2022-02-08T12:58:58Z"
  lastOperation:
    description: Reconciling the bastion
    lastUpdateTime: "2022-02-08T13:00:20Z"
    progress: 1
    state: Processing
    type: Reconcile
  observedGeneration: 2
```

```
{"level":"error","ts":"2022-02-08T13:57:24.356+0100","logger":"bastion_controller","msg":"error","namespace":"shoot--core--ps-bastion-test","name":"cli-az2","error":"Error reconciling bastion: can't update Network Security Group shoot--core--ps-bastion-test-workers: network.SecurityGroupsClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code=\"SecurityRuleInvalidAddressPrefix\" Message=\"Security rule /subscriptions/00d2caa5-cd29-46f7-845a-2f8ee0360ef5/resourceGroups/shoot--core--ps-bastion-test/providers/Microsoft.Network/networkSecurityGroups/shoot--core--ps-bastion-test-workers/securityRules/shoot--core--ps-bastion-test-cli--bastion-d379b-allow-ssh IPv4 has invalid Address prefix. Value provided: 1.2.3.4/32,3.4.5.6/32\" Details=[]","stacktrace":"github.com/gardener/gardener/extensions/pkg/controller/bastion.(*reconciler).Reconcile\n\t/Users/d050337/Downloads/gardener-extension-provider-azure-bastion-azure/gardener-extension-provider-azure/vendor/github.com/gardener/gardener/extensions/pkg/controller/bastion/reconciler.go:97\ngithub.com/gardener/gardener/pkg/controllerutils/reconciler.(*operationAnnotationWrapper).Reconcile\n\t/Users/d050337/Downloads/gardener-extension-provider-azure-bastion-azure/gardener-extension-provider-azure/vendor/github.com/gardener/gardener/pkg/controllerutils/reconciler/operation_annotation_wrapper.go:73\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/Users/d050337/Downloads/gardener-extension-provider-azure-bastion-azure/gardener-extension-provider-azure/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:114\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/Users/d050337/Downloads/gardener-extension-provider-azure-bastion-azure/gardener-extension-provider-azure/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:311\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/Users/d050337/Downloads/gardener-extension-provider-azure-bastion-azure/gardener-extension-provider-azure/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/Users/d050337/Downloads/gardener-extension-provider-azure-bastion-azure/gardener-extension-provider-azure/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:227"}
```

Reply:
I can confirm this is fixed